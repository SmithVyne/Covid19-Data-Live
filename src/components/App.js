import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Route, Switch, Link } from 'react-router-dom';
import defaults from '../utilities/defaults';
import ADD_DATA from '../actions';
import ContinentsList from '../containers/ContinentsList';
import CountriesList from '../containers/CountriesList';
import Filter from './Filter';
import '../styles/App.css';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      filter: '',
      category: '',
    };

    this.handleFilter = this.handleFilter.bind(this);
    this.handleCategory = this.handleCategory.bind(this);
  }

  componentDidMount() {
    const { ADD_DATA } = this.props;

    fetch(defaults['base-url'])
      .then(response => response.json())
      .then(data => ADD_DATA(data));
  }

  handleFilter({ value }) {
    this.setState({ filter: value });
  }

  handleCategory({ value }) {
    this.setState({ category: value });
  }

  render() {
    const { filter, category } = this.state;
    return (
      <>
        <nav>
          <input id="filter-field" type="text" onChange={e => this.handleFilter(e.target)} value={filter} placeholder="Country/Continent" />
          <select id="filter-category" onChange={e => this.handleCategory(e.target)}>
            <option disabled selected>Select</option>
            <option>Country</option>
            <option>Continent</option>
          </select>
          <Link to="/filter" id="filter-btn">Filter</Link>
        </nav>

        <main>
          <Switch>
            <Route exact path="/" component={ContinentsList} />
            <Route path="/countries/:continent" component={CountriesList} />
            <Route path="/filter">
              <Filter filter={filter} category={category} />
            </Route>
          </Switch>
        </main>
      </>
    );
  }
}

App.propTypes = {
  ADD_DATA: PropTypes.func.isRequired,
};

const mapStateToProps = ({ data }) => ({ data });

export default connect(
  mapStateToProps,
  { ADD_DATA },
)(App);
