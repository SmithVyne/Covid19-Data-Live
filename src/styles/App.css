@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,900&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
  font-family: Montserrat, serif;
}

html,
body,
#root {
  width: 100%;
  min-height: 100vh;
  max-height: max-content;
}

#root {
  background-color: #ec4c8a;
  padding-top: 10px;
}

nav {
  width: 100%;
  max-width: 800px;
  margin: auto;
  display: grid;
  grid-template-columns: 70% 20% 10%;
  grid-auto-rows: 100%;
  height: 30px;
}

#filter-field {
  padding: 0 23px;
  outline: none;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 4px;
  background-color: #d31860;
  font-weight: bold;
  font-size: 1em;
  text-align: center;
}

#filter-field::placeholder {
  color: rgba(255, 255, 255, 0.575);
  font-size: smaller;
}

#filter-category {
  cursor: pointer;
  outline: none;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 4px;
  background-color: #d31860;
  font-weight: bold;
}

#filter-btn {
  text-decoration: none;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  background-color: #d31860;
}

main {
  width: 100%;
  max-width: 800px;
  height: max-content;
  margin: auto;
  margin-top: 10px;
  color: #fff;
}

#continentsList {
  width: inherit;
  height: inherit;
  max-height: inherit;
}

.continent {
  width: 100%;
  height: calc(100vh / 8);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px 0 30px;
  text-decoration: none;
  color: #fff;
}

.continent div {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
  width: 170px;
  height: 80%;
  font-size: 14px;
  letter-spacing: 0.5px;
}




/* Loading Widget */
.lds-roller {
  display: block;
  position: relative;
  width: 80px;
  height: 80px;
  margin-top: 15%;
  margin-left: 45%;
}
.lds-roller div {
  animation: lds-roller 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  transform-origin: 40px 40px;
}
.lds-roller div:after {
  content: " ";
  display: block;
  position: absolute;
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #fff;
  margin: -4px 0 0 -4px;
}
.lds-roller div:nth-child(1) {
  animation-delay: -0.036s;
}
.lds-roller div:nth-child(1):after {
  top: 63px;
  left: 63px;
}
.lds-roller div:nth-child(2) {
  animation-delay: -0.072s;
}
.lds-roller div:nth-child(2):after {
  top: 68px;
  left: 56px;
}
.lds-roller div:nth-child(3) {
  animation-delay: -0.108s;
}
.lds-roller div:nth-child(3):after {
  top: 71px;
  left: 48px;
}
.lds-roller div:nth-child(4) {
  animation-delay: -0.144s;
}
.lds-roller div:nth-child(4):after {
  top: 72px;
  left: 40px;
}
.lds-roller div:nth-child(5) {
  animation-delay: -0.18s;
}
.lds-roller div:nth-child(5):after {
  top: 71px;
  left: 32px;
}
.lds-roller div:nth-child(6) {
  animation-delay: -0.216s;
}
.lds-roller div:nth-child(6):after {
  top: 68px;
  left: 24px;
}
.lds-roller div:nth-child(7) {
  animation-delay: -0.252s;
}
.lds-roller div:nth-child(7):after {
  top: 63px;
  left: 17px;
}
.lds-roller div:nth-child(8) {
  animation-delay: -0.288s;
}
.lds-roller div:nth-child(8):after {
  top: 56px;
  left: 12px;
}
@keyframes lds-roller {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
