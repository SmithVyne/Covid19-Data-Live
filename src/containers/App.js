import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Route, Switch, Link } from 'react-router-dom';
import defaults from '../utilities/defaults';
import ADD_DATA from '../actions';
import ContinentsList from './ContinentsList';
import CountriesList from './CountriesList';
import Filter from '../components/Filter';
import '../styles/App.css';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      filter: '',
      category: '',
      render: false,
    };

    this.handleFilter = this.handleFilter.bind(this);
    this.handleCategory = this.handleCategory.bind(this);
  }

  componentDidMount() {
    const { ADD_DATA } = this.props;

    fetch(defaults['base-url'])
      .then(response => response.json())
      .then(data => ADD_DATA(data) && this.setState({ render: true }));
  }

  handleFilter({ value }) {
    this.setState({ filter: value });
  }

  handleCategory({ value }) {
    this.setState({ category: value });
  }

  render() {
    const { filter, category, render } = this.state;
    if (render) {
      return (
        <>
          <nav>
            <input id="filter-field" type="text" onChange={e => this.handleFilter(e.target)} value={filter} placeholder="Country/Continent" />
            <select id="filter-category" onChange={e => this.handleCategory(e.target)}>
              <option>Country</option>
              <option>Continent</option>
            </select>
            <Link to="/filter" id="filter-btn">Filter</Link>
          </nav>

          <main>
            <Switch>
              <Route exact path="/" component={ContinentsList} />
              <Route path="/countries/:continent" component={CountriesList} />
              <Route path="/filter">
                <Filter filter={filter} category={category} />
              </Route>
            </Switch>
          </main>
        </>
      );
    }
    return (
      <div className="lds-roller">
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    );
  }
}

App.propTypes = {
  ADD_DATA: PropTypes.func.isRequired,
};

export default connect(
  null,
  { ADD_DATA },
)(App);
