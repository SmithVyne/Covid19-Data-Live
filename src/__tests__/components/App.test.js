import { render, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import App from '../../components/App';
import Reducer from '../../reducer';

describe('Tests the app functions', () => {
  const store = createStore(Reducer);

  const { getByRole, getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
  
  it('First displays the home page with a nav bar', async () => {
    await waitFor(() => {
      expect(getByRole('navigation')).toBeInTheDocument();
    });
  });
});
