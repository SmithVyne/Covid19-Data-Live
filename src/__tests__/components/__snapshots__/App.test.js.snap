// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the app functions Tests the filter component and route 1`] = `
<main>
  <div
    id="continentsList"
  />
</main>
`;
